cluster-autoscaler/simulator/clustersnapshot/clustersnapshot.go

```diff
@@ -29,24 +29,22 @@ import (
 // It exposes mutation methods and can be viewed as scheduler's SharedLister.
 type ClusterSnapshot interface {
 	schedulerframework.SharedLister
-	// AddNode adds node to the snapshot.
-	AddNode(node *apiv1.Node) error
-	// AddNodes adds nodes to the snapshot.
-	AddNodes(nodes []*apiv1.Node) error
-	// RemoveNode removes nodes (and pods scheduled to it) from the snapshot.
-	RemoveNode(nodeName string) error
-	// AddPod adds pod to the snapshot and schedules it to given node.
-	AddPod(pod *apiv1.Pod, nodeName string) error
-	// RemovePod removes pod from the snapshot.
-	RemovePod(namespace string, podName string, nodeName string) error
-	// AddNodeWithPods adds a node and set of pods to be scheduled to this node to the snapshot.
-	AddNodeWithPods(node *apiv1.Node, pods []*apiv1.Pod) error
-	// IsPVCUsedByPods returns if the pvc is used by any pod, key = <namespace>/<pvc_name>
-	IsPVCUsedByPods(key string) bool
+
+	// Initialize clears the snapshot and initializes it with real objects from the cluster - Nodes,
+	// scheduled pods.
+	Initialize(nodes []*apiv1.Node, scheduledPods []*apiv1.Pod) error
```

Comment:
I don't like the name "Initialize" here. I think it's misleading:
 * I'd expect "Initialize" to be something that happens one on the start of object's lifetime. 
 * I'd expect "Initialize" to be required before using the object in any other way - this isn't the case here, see my comment in one of the tests for example. 
 * It's very non-specific, literally anything can happen here. It's fine if it encapsulates a bunch of internal stuff I don't care about as a user, but here it performs a very specific function that is both understandable and important for the user. 

Naming is hard obviously, but I'd suggest the name along the lines of "ReplaceContentWith()" or "SetContent" or "SetClusterState" or similar. I think any of this would imply that the current contents of snapshot are dropped and replaced with whatever is provided - which is exactly what the comment above is saying.

Reply:
All good points, changed the name to `SetClusterState`, I think this leaves us with clear semantics:

`SetClusterState` resets the forks and sets the state to what's provided. No expectation of happening only once or having to be called before other methods, the name reflects what's happening.